src/components/CampaignLocations/CampaignLocationsScreen.tsx

```diff
@@ -0,0 +1,181 @@
+import React, {
+  FunctionComponent,
+  useContext,
+  useEffect,
+  useLayoutEffect,
+  useCallback
+} from "react";
+import { View, StyleSheet } from "react-native";
+import { size, fontSize } from "../../common/styles";
+import { Credits } from "../Credits";
+import { KeyboardAvoidingScrollView } from "../Layout/KeyboardAvoidingScrollView";
+import { TopBackground } from "../Layout/TopBackground";
+import { AppHeader } from "../Layout/AppHeader";
+import { ImportantMessageContentContext } from "../../context/importantMessage";
+import { Banner } from "../Layout/Banner";
+import { FeatureToggler } from "../FeatureToggler/FeatureToggler";
+import { HelpButton } from "../Layout/Buttons/HelpButton";
+import { HelpModalContext } from "../../context/help";
+import { AuthStoreContext } from "../../context/authStore";
+import { CampaignLocationsListItem } from "./CampaignLocationsListItem";
+import { NavigationProps, AuthCredentials } from "../../types";
+import { Sentry } from "../../utils/errorTracking";
+import { CampaignConfigsStoreContext } from "../../context/campaignConfigsStore";
+import { useDrawerContext } from "../../context/drawer";
+import { LoadingView } from "../Loading";
+import { Card } from "../Layout/Card";
+import { AppText } from "../Layout/AppText";
+
+const styles = StyleSheet.create({
+  content: {
+    position: "relative",
+    padding: size(2),
+    paddingVertical: size(8),
+    height: "100%",
+    width: 512,
+    maxWidth: "100%"
+  },
+  headerText: {
+    marginBottom: size(4)
+  },
+  bannerWrapper: {
+    marginBottom: size(1.5)
+  },
+  loadingViewWrapper: {
+    alignItems: "center",
+    justifyContent: "center"
+  },
+  selectCampaignHeader: {
+    fontFamily: "brand-bold",
+    fontSize: fontSize(1)
+  },
+  campaignLocationWrapper: {
+    marginHorizontal: -size(3),
+    marginTop: size(2)
+  }
+});
+
+export const CampaignLocationsScreen: FunctionComponent<NavigationProps> = ({
+  navigation
+}) => {
+  useEffect(() => {
+    Sentry.addBreadcrumb({
+      category: "navigation",
+      message: "CampaignLocationsScreen"
+    });
+  }, []);
+
+  const messageContent = useContext(ImportantMessageContentContext);
+  const showHelpModal = useContext(HelpModalContext);
+
+  const {
+    hasLoadedFromStore: hasLoadedAuthFromStore,
+    authCredentials
+  } = useContext(AuthStoreContext);
+  const {
+    hasLoadedFromStore: hasLoadedCampaignConfigFromStore,
+    allCampaignConfigs
+  } = useContext(CampaignConfigsStoreContext);
+
+  const { setDrawerButtons } = useDrawerContext();
+
+  useEffect(() => {
+    const drawerButtons = [
+      {
+        icon: "map-marker-plus",
+        label: "Add campaign",
+        onPress: () => {
+          navigation.navigate("LoginScreen");
+        }
+      }
+    ];
+
+    drawerButtons.push({
+      icon: "map-search",
+      label: "Change campaign",
+      onPress: () => {
+        navigation.navigate("CampaignLocationsScreen");
+      }
+    });
```

Comment:
Hmm, actually since we're showing this all the time, do we wanna add it to the array `drawerButtons` above?